File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/controller/book/request/ReturnBookRequest.java
package dev.starryeye.book_library.controller.book.request;

import dev.starryeye.book_library.application.facade.input.ReturnBookInput;
import jakarta.validation.constraints.NotNull;

public record ReturnBookRequest(
        @NotNull(message = "bookId must not be null")
        Long bookId,
        @NotNull(message = "userId must not be null")
        Long userId
) {

    public ReturnBookInput toInput() {
        return new ReturnBookInput(bookId, userId);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/request/ReturnBookRequest.kt
package dev.starryeye.book_library.controller.book.request

import dev.starryeye.book_library.application.facade.input.ReturnBookInput
import jakarta.validation.constraints.NotNull

data class ReturnBookRequest(
    @field:NotNull
    val bookId: Long?,
    @field:NotNull
    val userId: Long?,
)

fun ReturnBookRequest.toInput(): ReturnBookInput {
    return ReturnBookInput(
        bookId = bookId!!,
        userId = userId!!
    )
}